File: sdk/src/main/java/com/sygic/travel/sdk/geo/spread/SpreadedPlace.java
package com.sygic.travel.sdk.geo.spread;

import android.graphics.Point;

import com.sygic.travel.sdk.model.place.Place;

/**
 * <p>Wrapper of {@link Place}, which has been already processed by the spread algorithm.</p>
 */
public class SpreadedPlace {
	private Place place;
	private Point canvasCoords;
	private SpreadSizeConfig sizeConfig;

	/**
	 * <p>Wrapper of {@link Place}, which has been already processed by the spread algorithm.</p>
	 * @param place A processed place.
	 * @param canvasCoords Place's {@code x, y} coordinates on map's canvas on display.
	 * @param sizeConfig Place's {@link SpreadSizeConfig}.
	 */
	public SpreadedPlace(Place place, Point canvasCoords, SpreadSizeConfig sizeConfig) {
		this.place = place;
		this.canvasCoords = canvasCoords;
		this.sizeConfig = sizeConfig;
	}

	public Place getPlace() {
		return place;
	}

	public void setPlace(Place place) {
		this.place = place;
	}

	public Point getCanvasCoords() {
		return canvasCoords;
	}

	public void setCanvasCoords(Point canvasCoords) {
		this.canvasCoords = canvasCoords;
	}

	public SpreadSizeConfig getSizeConfig() {
		return sizeConfig;
	}

	public void setSizeConfig(SpreadSizeConfig sizeConfig) {
		this.sizeConfig = sizeConfig;
	}
}

File: sdk/src/main/java/com/sygic/travel/sdk/geo/spread/SpreadedPlace.kt
package com.sygic.travel.sdk.geo.spread

import android.graphics.Point

import com.sygic.travel.sdk.model.place.Place

/**
 *
 * Wrapper of [Place], which has been already processed by the spread algorithm.
 */
class SpreadedPlace
/**
 *
 * Wrapper of [Place], which has been already processed by the spread algorithm.
 * @param place A processed place.
 * *
 * @param canvasCoords Place's `x, y` coordinates on map's canvas on display.
 * *
 * @param sizeConfig Place's [SpreadSizeConfig].
 */
(var place: Place?, var canvasCoords: Point?, var sizeConfig: SpreadSizeConfig?)